org.lflang/src/org/lflang/federated/extensions/TSExtension.java

```diff
@@ -0,0 +1,163 @@
+package org.lflang.federated.extensions;
+
+import static org.lflang.util.StringUtil.addDoubleQuotes;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.stream.Collectors;
+
+import org.lflang.ASTUtils;
+import org.lflang.ErrorReporter;
+import org.lflang.InferredType;
+import org.lflang.Target;
+import org.lflang.TargetProperty.CoordinationType;
+import org.lflang.TimeValue;
+import org.lflang.federated.generator.FedASTUtils;
+import org.lflang.federated.generator.FedConnectionInstance;
+import org.lflang.federated.generator.FedFileConfig;
+import org.lflang.federated.generator.FederateInstance;
+import org.lflang.generator.GeneratorBase;
+import org.lflang.generator.LFGeneratorContext;
+import org.lflang.generator.ReactorInstance;
+import org.lflang.generator.ts.TSExtensionsKt;
+import org.lflang.lf.Action;
+import org.lflang.lf.Output;
+import org.lflang.lf.VarRef;
+import org.lflang.lf.Variable;
+
+public class TSExtension implements FedTargetExtension {
+    @Override
+    public void initializeTargetConfig(LFGeneratorContext context, int numOfFederates, FederateInstance federate, FedFileConfig fileConfig, ErrorReporter errorReporter, LinkedHashMap<String, Object> federationRTIProperties) throws IOException {
+
+    }
+
+    @Override
+    public String generateNetworkReceiverBody(Action action, VarRef sendingPort, VarRef receivingPort, FedConnectionInstance connection, InferredType type, CoordinationType coordinationType, ErrorReporter errorReporter) {
+        return """
+        // generateNetworkReceiverBody
+        if (%1$s !== undefined) {
+            %2$s.%3$s = %1$s;
+        }
+        """.formatted(
+            action.getName(),
+            receivingPort.getContainer().getName(),
+            receivingPort.getVariable().getName()
+        );
+    }
+
+    @Override
+    public String generateNetworkSenderBody(VarRef sendingPort, VarRef receivingPort, FedConnectionInstance connection, InferredType type, CoordinationType coordinationType, ErrorReporter errorReporter) {
+        return"""
+        if (%1$s.%2$s !== undefined) {
+            this.util.sendRTITimedMessage(%1$s.%2$s, %3$s, %4$s);
+        }
+        """.formatted(
+            sendingPort.getContainer().getName(),
+            sendingPort.getVariable().getName(),
+            connection.getDstFederate().id,
+            connection.getDstFederate().networkMessageActions.size()
+        );
+    }
+
+    @Override
+    public String generateNetworkInputControlReactionBody(int receivingPortID, TimeValue maxSTP, CoordinationType coordination) {
+        return "// TODO(hokeun): Figure out what to do for generateNetworkInputControlReactionBody";
```

Comment:
there are still a couple of todos here

Reply:
@hokeun?

Reply:
If I remember correctly, these were added by @Soroosh129 as placeholders when he first created this file. I think we had some discussion on control reaction bodies with @lhstrh and @byeong-gil last fall (or summer), but I can't remember right now. Let me think about it again...

Reply:
Ah, I remembered this. These are related to port absent messages, and we currently don't have any test requiring port absent yet. Thus, we can leave this as is for now. I think we first need to have the port absent implementation in reactor-ts merged, like these two : https://github.com/lf-lang/reactor-ts/pull/119, https://github.com/lf-lang/reactor-ts/pull/117

Reply:
Here's ongoing PR for adding: NetworkOutputControlReactionBody https://github.com/lf-lang/lingua-franca/pull/1333/files I'll work with @byeong-gil to merge this PR soon.